functions_complete.py

```diff
@@ -0,0 +1,310 @@
+        
+        grid[y_to_fill, x_to_fill] = 1
+        # Is there a way to name the ship that I'm currently using?
+        print("You just placed the", ships[ship])
+        print("Current grid:")
+        print(grid)
+    
+    # Now remove the first and last columns and rows because they were only included to allow the check_position function
+    grid = grid[1:11, 1:11]
+    print("Final grid:")
+    print(grid)
+    
+    return (grid)
+
+######## Then the player and the computer take turns at shooting each other
+def players_shot(visible_grid_for_player, computers_ships_uncovered):
+    """
+    This function should allow the player and the computer to shoot at each other.
+
+    Parameters
+    ----------
+    visible_grid_for_player : TYPE
+        DESCRIPTION.
+    computers_ships_uncovered : TYPE
+        DESCRIPTION.
+
+    Returns
+    -------
+    None.
+
+    """
+    guess_y = int(input("Enter which row you want to shoot at (a value between 0 and 9): "))
+    guess_x = int(input("Enter which column you want to shoot at (a value between 0 and 9): "))
+    
+    if visible_grid_for_player[guess_y, guess_x] != 0:
+        print("This point is not unknown. Please choose an unknown point")
+        players_turn = True
+        
+        return(players_turn, visible_grid_for_player, computers_ships_uncovered) # start over
+    
+    elif visible_grid_for_player[guess_y, guess_x] == 0:
+        visible_grid_for_player[guess_y, guess_x] = 999
+        print("This is the position you selected:")
+        print(visible_grid_for_player)
+        
+        # confirm = input("Enter 'yes' to confirm, enter 'no' to enter new input")
+        # if (confirm == "yes"):
+        #     break
+        # elif (confirm == "no"):
+        #     players_turn = True
+        # return(players_turn, visible_grid_for_player) # start over
+    
+    if (computers_grid[guess_y, guess_x] == 1):
+        visible_grid_for_player[guess_y, guess_x] = 1
+        print("Your shot was succesful. This is the computer's grid currently visible to you:")
+        print(visible_grid_for_player)
+        computers_ships_uncovered.append(True)
+        
+        players_turn = True
+        
+    elif (computers_grid[guess_y, guess_x] != 1):
+        visible_grid_for_player[guess_y, guess_x] = 2
+        print("Your shot was unsuccesful. This is the computer's grid currently visible to you:")
+        print(visible_grid_for_player)
+        
+        players_turn = False
+        
+    return(players_turn, visible_grid_for_player, computers_ships_uncovered)
+
+def computers_shot(visible_grid_for_computer, players_ships_uncovered):
+    guess_y = random.randrange(0, 10, 1)
+    guess_x = random.randrange(0, 10, 1)
+    
+    if visible_grid_for_computer[guess_y, guess_x] != 0:
+        
+        players_turn = False
+        
+        return(players_turn, visible_grid_for_computer) # computer will start over
+    
+    if (players_grid[guess_y, guess_x] == 1):
```

Comment:
I'd consider putting some kind of await statement in here.
Right now the computer's turn will be instantaneous and as soon as your turn is done, it will be your turn again. Especially, if they keep hitting you and do multiple turns in a row, it feels weird to have those unfold within a split second.
Maybe include a half-second or second break in there, somewhere